src/main.ts

```diff
@@ -31,6 +32,17 @@ function initialize() {
   createPreviewWindow();
 }
 
+function getMainWindowDisplay() {
+  if (mainWindow) {
```

Comment:
I struggled so hard to not implicit cast objects to booleans 😭
Btw I've put all the utility code at the bottom to keep this file organized. I've felt splitting would just make things worse though.

Reply:
😅 I don't really see what's wrong with this though, it's even a little safer since it'll work for both undefined and null
But okay, I'll clean it up!

Reply:
there is an eslint rule for this but we currently get 100+ errors..
https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/strict-boolean-expressions.md

Reply:
I already fixed the issue with undefined by setting the initial value to null. Now the code should be sound in terms of type correctness. It's not wrong but I think it is more readable or at least I do not need to remember what the type of the variable is to understand the code. For example, it disrupts my flow when I see `!someValue.length` as the length has to be cast to a boolean and then negated even though all it does is to check if a string is not empty.

Reply:
Or `!someValue` is a number and not a boolean and the intent was to check if the value is 0. By the time I figured it out it would have been faster to write it out.